nvme.c

```diff
@@ -925,22 +929,24 @@ int __id_ctrl(int argc, char **argv, struct command *cmd, struct plugin *plugin,
 	if (cfg.human_readable)
 		flags |= HUMAN;
 
-	err = nvme_identify_ctrl(fd, &ctrl);
-	if (!err) {
+	err = nvme_ctrl_get_by_fd(fd, &ctrl, &err_msg);
```

Comment:
The function `nvme_ctrl_get_by_fd` will allocate a string on the heap for `err_msg` to return it.  This function doesn't print the value or free the value?